.github/workflows/_rust.yml

```diff
@@ -56,8 +56,8 @@ jobs:
         timeout-minutes: 5
       - uses: taiki-e/install-action@cargo-udeps
       - run: |
-          rustup install --no-self-update nightly-2024-03-26 --profile minimal # The exact nightly version doesn't matter, just pin a random one.
-          cargo +nightly-2024-03-26 udeps --all-targets --all-features ${{ steps.setup-rust.outputs.packages }}
+          rustup install --no-self-update nightly-2024-06-01 --profile minimal # The exact nightly version doesn't matter, just pin a random one.
+          cargo +nightly-2024-06-01 udeps --all-targets --all-features ${{ steps.setup-rust.outputs.packages }}
```

Comment:
I tried a nightly from July but it threw some strange error about `!` not decaying to `()` which I couldn't understand and couldn't fix. It's related to this issue:

https://github.com/rust-lang/rust/issues/123748